examples/llm_plus_gnn/glem.py

```diff
@@ -0,0 +1,372 @@
+        nn = 1
+        for s in list(p.size()):
+            nn = nn * s
+        pp += nn
+    return pp
+
+
+def main(args):
+    gpu = args.gpu
+    dataset_name = args.dataset
+    root = osp.join(parent_dir, 'data', 'ogb')
+    hf_model = args.hf_model
+    pl_ratio = args.pl_ratio
+    gnn_lr = args.gnn_lr
+    lm_lr = args.lm_lr
+    em_order = args.em_order
+    gnn_epochs = args.gnn_epochs
+    lm_epochs = args.lm_epochs
+    patience = args.patience
+    verbose = args.verbose
+    out_dir = args.out_dir
+    lm_batch_size = args.lm_batch_size
+    gnn_batch_size = args.gnn_batch_size
+    lm_use_lora = args.lm_use_lora
+    token_on_disk = args.token_on_disk
+    num_em_iters = args.num_em_iters
+    args.external_pred_path
+    start_time = time.time()
+
+    device = torch.device(
+        f'cuda:{gpu}' if torch.cuda.is_available() else 'cpu')
+    print(f'Running on: {torch.cuda.get_device_name({gpu})}')
+    torch.cuda.empty_cache()
+
+    from ogb.nodeproppred import PygNodePropPredDataset
+    dataset = PygNodePropPredDataset(f'ogbn-{dataset_name}', root=root)
+    split_idx = dataset.get_idx_split()
+    data = dataset.data
+
+    tag_dataset = TAGDataset(root, dataset, hf_model,
+                             token_on_disk=token_on_disk)
+    text_dataset = tag_dataset.to_text_dataset()
+    print(tag_dataset.num_classes, tag_dataset.raw_file_names)
+
+    num_classes = tag_dataset.num_classes
+    num_features = data.num_features
+    # =========================== LM Data split ===============================
+    split_idx = tag_dataset.get_idx_split()
+
+    # GLEM train with augmented data, mark original train data as gold data,
+    gold_idx = split_idx['train']
+    split_idx['valid']
+    test_idx = split_idx['test']
+
+    # randome sample pseudo labels nodes, generate their index
+    num_pseudo_labels = int(gold_idx.numel() * pl_ratio)
+    idx_to_select = torch.randperm(test_idx.numel())[:num_pseudo_labels]
+    pseudo_labels_idx = test_idx[idx_to_select]
+    train_idx = torch.cat(
+        (gold_idx, pseudo_labels_idx))  # augmented train_indx
+
+    print(f'train_idx: {train_idx.size(0)}, '
+          f'gold_idx: {gold_idx.size(0)}, '
+          f'pseudo labels ratio: {pl_ratio}, '
+          f'{train_idx.size(0)/gold_idx.size(0) - 1.0}')
+    gold_dataset = torch.utils.data.Subset(dataset=text_dataset,
+                                           indices=gold_idx)
+    train_dataset = torch.utils.data.Subset(dataset=text_dataset,
+                                            indices=train_idx)
+    # valid_dataset = torch.utils.data.Subset(dataset=text_dataset,
+    #                                         indices=valid_idx)
+
+    # ========================== LM Data Loader ===============================
+
+    print('Building language model dataloader...', end='-->')
+    from torch_geometric.loader import DataLoader
+    text_pretrain_loader = DataLoader(gold_dataset, batch_size=lm_batch_size,
```

Comment:
would be good to check what happens if you set pin_memory=True, if makes faster, would be worth turning it on

Reply:
pin_memory is a argument for `torch.utils.data.DataLoader` not include in `torch_geometric.loader.DataLoader` I have remove it in current example

Reply:
actually, It does not make any difference for speed after testing

Reply:
okay lets leave it removed then